apm-agent-plugins/apm-spring-resttemplate/apm-spring-restemplate-plugin/src/main/java/co/elastic/apm/agent/resttemplate/SpringRestTemplateAdvice.java

```diff
@@ -46,11 +47,22 @@
     @Advice.OnMethodEnter(suppress = Throwable.class, inline = false)
     public static Object beforeExecute(@Advice.This ClientHttpRequest request) {
         logger.trace("Enter advice for method {}#execute()", request.getClass().getName());
-        if (TracerAwareInstrumentation.tracer.getActive() == null) {
+
+        final AbstractSpan<?> parent = TracerAwareInstrumentation.tracer.getActive();
+        if (parent == null) {
             return null;
         }
-        final AbstractSpan<?> parent = TracerAwareInstrumentation.tracer.getActive();
-        Span span = HttpClientHelper.startHttpClientSpan(parent, Objects.toString(request.getMethod()), request.getURI(), request.getURI().getHost());
+        URI uri = null;
+        String host = "unknown";
+        try {
+            uri = request.getURI();
+        } catch (NoSuchMethodError ignored) {
```

Comment:
Better to make sure we don't instrument Spring RestTemplate < 3.0.3

Reply:
fixed in the last commit.